src/Poseidon/Janno.hs

```diff
@@ -394,15 +395,41 @@ createMinimalSample (EigenstratIndEntry id_ sex pop) =
 
 -- Janno file writing
 
+-- | A helper functions to replace empty bytestrings values in janno files with explicit "n/a"
+explicitNA :: Bch.ByteString -> Bch.ByteString
+explicitNA = replaceInJannoBytestring Bch.empty "n/a"
+
+replaceInJannoBytestring :: Bch.ByteString -> Bch.ByteString -> Bch.ByteString -> Bch.ByteString
+replaceInJannoBytestring from to tsv =
+    let tsvRows = Bch.lines tsv
+        tsvCells = map (Bch.splitWith (=='\t')) tsvRows
+        tsvCellsUpdated = map (map (\y -> if y == from || y == Bch.append from "\r" then to else y)) tsvCells
```

Comment:
What's going on with "\r"? Why would that happen?

Reply:
No idea - I didn't touch this code beyond moving it around. I'll have a look.

Reply:
I managed to remove this function entirely.